src/Datagrid/ProxyQuery.php

```diff
@@ -197,7 +197,10 @@ public function getDoctrineQuery(): Query
     public function setSortBy(array $parentAssociationMappings, array $fieldMapping): BaseProxyQueryInterface
     {
         $alias = $this->entityJoin($parentAssociationMappings);
-        $this->sortBy = $alias.'.'.$fieldMapping['fieldName'];
+
+        $this->sortBy = isset($fieldMapping['fieldName']) ?
```

Comment:
if we do like this, I can call setSortBy([], []) to reset the sort (It still won't reset the entityJoin, that would require a different approach). TBH I am not sure why a sortBy does and entityJoin that is not easy to undo under the hood.

WDYT about this approach @VincentLanglet ?

Reply:
This seems a good idea.

I would slighty change the implementation to
```
if (!isset($fieldMapping['fieldName'])) {
    $this->sortBy = null;
    return $this;
}

$alias = $this->entityJoin($parentAssociationMappings);
$this->sortBy = $alias.'.'.$fieldMapping['fieldName'];
return $this;
```
and a test could be nice.

For the entityJoin, we have the call
```
$this->query->setSortBy(
            $this->values[DatagridInterface::SORT_BY]->getSortParentAssociationMapping(),
            $this->values[DatagridInterface::SORT_BY]->getSortFieldMapping()
        );
```
in SonataAdmin, and the entityJoin is not in the interface, so it has to be in the sortby, no ?

Reply:
Added test to it.